ui/src/app/pages/homepage/data-access-requirements.tsx

```diff
@@ -0,0 +1,531 @@
+const bypassedOrCompletedText = (status: AccessModuleStatus) => {
+  const {completionEpochMillis, bypassEpochMillis}: AccessModuleStatus = status || {};
+  const userCompletedModule = !!completionEpochMillis;
+  const userBypassedModule = !!bypassEpochMillis;
+
+  return cond(
+      [userCompletedModule, () => `Completed on: ${displayDateWithoutHours(completionEpochMillis)}`],
+      [userBypassedModule, () => `Bypassed on: ${displayDateWithoutHours(bypassEpochMillis)}`],
+      // return nothing if there's no text
+    () => null
+  );
+};
+
+const handleTerraShibbolethCallback = (token: string, spinnerProps: WithSpinnerOverlayProps) => {
+  const handler = withErrorModal({
+    title: 'Error saving NIH Authentication status.',
+    message: 'An error occurred trying to save your NIH Authentication status. Please try again.',
+    onDismiss: () => {
+      spinnerProps.hideSpinner();
+    }
+  })(async() => {
+    spinnerProps.showSpinner();
+    await profileApi().updateNihToken({jwt: token});
+    spinnerProps.hideSpinner();
+  });
+
+  return handler();
+};
+
+const handleRasCallback = (code: string, spinnerProps: WithSpinnerOverlayProps) => {
+  const handler = withErrorModal({
+    title: 'Error saving RAS Login.Gov linkage status.',
+    message: 'An error occurred trying to save your RAS Login.Gov linkage status. Please try again.',
+    onDismiss: () => {
+      spinnerProps.hideSpinner();
+    }
+  })(async() => {
+    spinnerProps.showSpinner();
+    await profileApi().linkRasAccount({ authCode: code, redirectUrl: buildRasRedirectUrl() });
+    spinnerProps.hideSpinner();
+  });
+
+  return handler();
+};
+
+interface ModuleProps {
+  module: AccessModule;
+  active: boolean;    // is this the currently-active module that the user should complete
+
+  // TODO RW-7059
+  // eligible: boolean;  // is the user eligible to complete this module (does the inst. allow it)
+}
+const MaybeModule = (props: ModuleProps): JSX.Element => {
+  const {profile} = useStore(profileStore);
+  const [navigate, ] = useNavigation();
+
+  const {module, active} = props;
+  const statusTextMaybe = bypassedOrCompletedText(getAccessModuleStatusByName(profile, module));
+
+  // whether to show the refresh button: this module has been clicked
+  const [showRefresh, setShowRefresh] = useState(false);
+
+  // whether to show the Two Factor Auth Modal
+  const [showTwoFactorAuthModal, setShowTwoFactorAuthModal] = useState(false);
+
+  const registrationTask = getRegistrationTask(navigate, module);
+
+  // kluge until we have fully migrated from the Registration Dashboard:
+  // getRegistrationTask() has onClick() functions for every module, which is generally what we want
+  // but we pop up a modal for Two Factor Auth instead of using the standard task
+  const moduleAction = registrationTask && (module === AccessModule.TWOFACTORAUTH ?
+      () => setShowTwoFactorAuthModal(true) :
+      registrationTask.onClick);
+
+  const Refresh = () => <Button
```

Comment:
Ah, is this the "closurized props" concern you mention below?  It seems to work but I certainly could be missing something here.

Reply:
after some googling, general consensus seems to be to avoid this pattern, probably safest is to move these definitions outside of this function component or turn these into helper functions rather than normal React components. 

https://stackoverflow.com/a/59636503/126626
https://stackoverflow.com/questions/59776799/its-okay-to-declare-other-components-inside-a-component
https://dev.to/borasvm/react-create-component-inside-a-component-456b

Honestly, none of those posts provide very convincing rationale though, I think I would need to spend a bit more time understanding React's rendering cycle to prove this to myself. Minimally, I think we can say that one side-effect of using this pattern:
ALL child components declared this way are rerendered on a state/prop change, where-as normally only children who have a state/prop change are rerendered.

However, I don't know the delta between that, and defining an inline helper function, which would have the same consequence. I might suggest checking with Isaac on this.

Reply:
If there's a potential cost to these sub-components, then we should probably avoid them in cases where there is dubious benefit - so I have now made several which don't use these props into independent components.

There is benefit to sharing props/state among Module, MaybeModule, and ModuleBox, so they continue to do so for now.